(module gben.tests.ErrorHandling)

(println "This is error handling test!")

;;; try is to put exception handler on expected failure point
(def result (try (/ 3 0)
             (recover e (do (println (:message e))
                            (println (:type e))
                            (println (:stack-trace e))
                            return 10))))

(assert-equals "result should be recovered with 10" result 10)
